### DISCUSSION TITLE
Enabling sections for specific filetypes

### DISCUSSION THREAD

**Discussion by RayZ0rr** [famiu/feline.nvim#206] [Q&A] [answered]
I am trying to get [asyncrun](https://github.com/skywind3000/asyncrun.vim) status in quickfix window statusline. However, I am not able to get it with following code. How should I modify the `enabled` part to detect a specific filetype?

```lua
components.inactive[2][1] = {
  provider = "Code Status:" .. '%{g:asyncrun_status}',
  enabled = vim.fn.exists('g:asyncrun_status') and print(vim.bo.filetype) == 'qf',
  hl = {
    fg = 'black',
  bg = 'cyan',
  style = 'bold'
  },
  left_sep = {
    str = ' ',
    hl = {
      fg = 'NONE',
      bg = 'cyan'
    }
  },
right_sep = {
  {
    str = ' ',
    hl = {
      fg = 'NONE',
      bg = 'cyan'
    }
  },
  ' '
}
}


```

**Comment by lukas-reineke** [comment, marked as answer]
The way you have it, it gets evaluated just once when you start Neovim. Instead, you want to make `enabled` a function, that returns a boolean.
Also, your usage of `print` is wrong, and you don't need to use `vim.fn.exists`

```lua
enabled = function()
    return vim.g.asyncrun_status ~= nil and vim.bo.filetype == 'qf'
end
```
